modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb

```diff
@@ -301,6 +301,96 @@
+                end
+              end
+            end
+          end
+        end
+
+        context 'when the country is invalid' do
+          let(:country) { 'United States of Nada' }
+
+          it 'responds with bad request' do
+            with_okta_user(scopes) do |auth_header|
+              VCR.use_cassette('evss/claims/claims') do
+                VCR.use_cassette('brd/countries') do
+                  json = JSON.parse(data)
+                  json['data']['attributes']['veteranIdentification']['mailingAddress']['country'] = country
+                  data = json.to_json
+                  post path, params: data, headers: headers.merge(auth_header)
+                  expect(response).to have_http_status(:bad_request)
+                end
+              end
+            end
+          end
+        end
+
+        context 'when no mailing address data is found' do
+          it 'responds with bad request' do
+            with_okta_user(scopes) do |auth_header|
+              VCR.use_cassette('evss/claims/claims') do
+                VCR.use_cassette('brd/countries') do
+                  json = JSON.parse(data)
+                  json['data']['attributes']['veteranIdentification']['mailingAddress'] = {}
+                  data = json.to_json
+                  post path, params: data, headers: headers.merge(auth_header)
+                  expect(response).to have_http_status(:unprocessable_entity)
+                end
+              end
+            end
+          end
+        end
+      end
+
+      context 'when the phone has non-digits included' do
+        let(:telephone) { '123456789X' }
+
+        it 'responds with bad request' do
+          with_okta_user(scopes) do |auth_header|
+            VCR.use_cassette('evss/claims/claims') do
+              VCR.use_cassette('brd/countries') do
+                json = JSON.parse(data)
+                json['data']['attributes']['veteranIdentification']['veteranNumber']['telephone'] = telephone
+                data = json.to_json
+                post path, params: data, headers: headers.merge(auth_header)
+                expect(response).to have_http_status(:unprocessable_entity)
+              end
+            end
+          end
+        end
+      end
+
+      context 'when the zipFirstFive has non-digits included' do
+        let(:zip_first_five) { '1234X' }
+
+        it 'responds with bad request' do
+          with_okta_user(scopes) do |auth_header|
+            VCR.use_cassette('evss/claims/claims') do
+              VCR.use_cassette('brd/countries') do
+                json = JSON.parse(data)
+                json['data']['attributes']['veteranIdentification']['mailingAddress']['zipFirstFive'] = zip_first_five
+                data = json.to_json
+                post path, params: data, headers: headers.merge(auth_header)
+                expect(response).to have_http_status(:unprocessable_entity)
+              end
+            end
+          end
+        end
+      end
```

Comment:
a few of the attributes do not have tests, do you mind adding some for them? apartmentOrUnitNumber, numberAndStreet, city, state, vaFileNumber, currentlyVaEmployee, serviceNumber, agreeToEmailRelatedToClaim, email

Reply:
Added tests for all attributes.